components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,340 @@
+        }
+    }
+
+    @Override
+    public List<LocalAuthenticatorConfig> getAllUserDefinedLocalAuthenticator(Integer tenantId)
+            throws AuthenticatorMgtException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+             NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                Query.GET_ALL_USER_DEFINED_AUTHENTICATOR_SQL)) {
+
+            statement.setString(Column.DEFINED_BY, DefinedByType.USER.toString());
+            statement.setInt(Column.TENANT_ID, tenantId);
+
+            List<LocalAuthenticatorConfig> allUserDefinedLocalConfigs = new ArrayList<>();
+            try (ResultSet rs = statement.executeQuery()) {
+                while (rs.next()) {
+                    LocalAuthenticatorConfig config;
+                    if (AuthenticationType.VERIFICATION.toString().equals(rs.getString(Column.AUTHENTICATION_TYPE))) {
+                        config = new VerificationAuthenticatorConfig();
+                    } else {
+                        config = new LocalAuthenticatorConfig();
+                    }
+                    config.setName(rs.getString(Column.NAME));
+                    config.setDisplayName(rs.getString(Column.DISPLAY_NAME));
+                    config.setEnabled(rs.getString(Column.IS_ENABLED).equals(IS_TRUE_VALUE));
+                    config.setDefinedByType(DefinedByType.valueOf(rs.getString(Column.DEFINED_BY)));
+                    allUserDefinedLocalConfigs.add(config);
+                }
+            }
+
+            return allUserDefinedLocalConfigs;
+        } catch (SQLException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(String.format("Error while retrieving the user defined local Authenticator for " +
+                                "tenant domain %s", IdentityTenantUtil.getTenantDomain(tenantId)));
+            }
+            ErrorMessages error = ErrorMessages.ERROR_WHILE_DELETING_AUTHENTICATOR;
+            throw new AuthenticatorMgtServerException(error.getMessage(), error.getMessage(), error.getCode(), e);
+        }
+    }
+
+    @Override
+    public void deleteUserDefinedLocalAuthenticator(String authenticatorConfigName, Integer tenantId)
+            throws AuthenticatorMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try (NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                Query.DELETE_AUTHENTICATOR_SQL)) {
+
+            statement.setString(Column.NAME, authenticatorConfigName);
+            statement.setInt(Column.TENANT_ID, tenantId);
+            statement.executeUpdate();
+
+            IdentityDatabaseUtil.commitTransaction(dbConnection);
+        } catch (SQLException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(String.format("Error while deleting the Authenticator of named: %s in Tenant Domain: %s. " +
```

Comment:
Fix the debug message as per the comment above

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8